v1/status_message.go

```diff
@@ -144,6 +154,13 @@ func (m *StatusMessage) HandleApplication() error {
 		return err
 	}
 	m.ParsedMessage = value
-
+	switch m.ParsedMessage.(type) {
```

Comment:
Do we have unit tests for `HandleApplication` where we could verify `MessageType` is set?

Reply:
no good point, I will add those